src/main/java/com/google/step/finscholar/data/Scholarship.java

```diff
@@ -0,0 +1,310 @@
+  public Optional<String> getAmountPerYear() {
+    return this.amountPerYear;
+  }
+
+  /** 
+   * The getter of the list of ethnicity and race requirements. 
+   * @return The list of ethnicity and race restrictions.
+   */
+  public List<String> getEthnicityRaceRequirements() {
+    return this.ethnicityRaceRequirements;
+  }
+
+  /**
+   * The getter of the list of financial background requirements.
+   * @return The list of financial requirements.
+   */
+  public List<String> getFinancialRequirements() {
+    return this.financialRequirements;
+  }
+
+  /**
+   * The getter for gender restrictions of the scholarship.
+   * @return The list of gender requirements.
+   */
+  public List<String> getGenderRequirements() {
+    return this.genderRequirements;
+  }
+
+  /**
+   * The getter for geographic restrictions of the scholarship;
+   * @return The list of location requirements.
+   */
+  public List<String> getLocationRequirements() {
+    return this.locationRequirements;
+  }
+
+  /**
+   * The getter for nationality restrictions of the scholarship;
+   * @return The list of national origin requirements.
+   */
+  public List<String> getNationalOriginRequirements() {
+    return this.nationalOriginRequirements;
+  }
+
+  /**
+   * The getter of the number of years the scholarship is offered to one student.
+   * @return The duration of the scholarship in years.
+   */
+  public Optional<Integer> getNumberOfYears() {
+    return this.numberOfYears;
+  }
+
+  /**
+   * The getter of the list of other requirements.
+   * @return The list of other requirements.
+   */
+  public List<String> getOtherRequirements() {
+    return this.otherRequirements;
+  }
+
+  /**
+   * The getter for the scholarship's name.
+   * @return The name of the scholarship.
+   */
+  public String getScholarshipName() {
+    return this.scholarshipName;
+  }
+
+  /**
+   * The getter for the UUID.
+   * @return the UUID of the scholarship.
+   */
+  public UUID getScholarshipUUID() {
+    return this.scholarshipUUID;
+  }
+  
+  /**
+   * The getter for the list of schools providing the scholarship.
+   * @return The schools list.
+   */
+  public List<UUID> getSchoolsList() {
+    return this.schoolsList;
+  }
+
+  /**
+   * The getter of the URL to the official website for the scholarship.
+   * @return The URL.
+   */
+  public String getURL() {
+    return this.URL;
+  }
+
+  /**
+   * The getter of the introduction.
+   * @return The optional containing introduction.
+   */
+  public Optional<String> getIntroduction() {
+    return this.introduction;
+  }
+
+  /**
+   * The getter of isRenewable.
+   * @return The Optional of boolean for isRenewable.
+   */
+  public Optional<Boolean> getIsRenewable() {
+    return this.isRenewable;
+  }
+
+  /**
+   * The setter for academicRequirements.
+   * @param academicRequirements
+   * @return The updated scholarship object.
+   */
+  public Scholarship setAcademicRequirements(List<String> academicRequirements) {
```

Comment:
Implement the builder pattern properly. Michael just did this for the College object in PR #17 if you need an example.

https://dzone.com/articles/design-patterns-the-builder-pattern

Reply:
Thanks, I just changed to the new pattern!